### ISSUE TITLE
Traffic prediction with advanced Graph Neural Networks By: DeepMind

### ISSUE THREAD

**Issue by QiXuanWang** [QiXuanWang/LearningFromTheBest#46]
Link: https://deepmind.com/blog/article/traffic-prediction-with-advanced-graph-neural-networks
Ref: https://www.reddit.com/r/MachineLearning/comments/impvog/r_deepmind_uses_gnns_to_boost_google_maps_eta/

Comment:
"
The ever-industrious DeepMind researchers meanwhile have been working on further improving Google Maps, and this week the UK-based AI company and research lab unveiled a partnership with Google Maps that has leveraged advanced Graph Neural Networks (GNNs) to improve estimated time of arrival (ETA) accuracy. The coordinated efforts have boosted the accuracy of real-time ETAs by up to 50 percent in cities such as Berlin, Jakarta, São Paulo, Sydney, Tokyo and Washington DC."

How:
1. Dividing the world’s roads into Supersegments consisting of multiple adjacent segments of road that share significant traffic volume
2. Novel Architectures:
  The biggest challenge to solve when creating a machine learning system to estimate travel times using Supersegments is an architectural one. How do we represent dynamically sized examples of connected segments with arbitrary accuracy in such a way that a single model can achieve success?
   In a Graph Neural Network, a message passing algorithm is executed where the messages and their effect on edge and node states are learned by neural networks. From this viewpoint, our Supersegments are road subgraphs, which were sampled at random in proportion to traffic density. A single model can therefore be trained using these sampled subgraphs, and can be deployed at scale.
  Each Supersegment, which can be of varying length and of varying complexity - from simple two-segment routes to longer routes containing hundreds of nodes - can nonetheless be processed by the same Graph Neural Network model. 

3. After much trial and error, however, we developed an approach to solve this problem by adapting a novel reinforcement learning technique for use in a supervised setting.
4. We found use of a linear combination of multiple loss functions (weighted appropriately) greatly increased the ability of the model to generalize.
